### PR TITLE
Test SMB synced folders

### PR DESCRIPTION
**Pull request by soapy1**

### CODE REVIEW

File: acceptance/support-skeletons/synced_folder_smb/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Vagrant.configure("2") do |config|
+  config.vm.box = "box"
+
+  config.vm.synced_folder ".", "/vagrant-smb", type: "smb", smb_username: ENV["SMB_USERNAME"], smb_password: ENV["SMB_PASSWORD"]
```
Comment (jbonhag) [inline acceptance/support-skeletons/synced_folder_smb/Vagrantfile:4]: I'd suggest naming these environment variables with a `VAGRANT_` prefix so it's clearer how they're being used.

```suggestion
  config.vm.synced_folder ".", "/vagrant-smb", type: "smb", smb_username: ENV["VAGRANT_SMB_USERNAME"], smb_password: ENV["VAGARANT_SMB_PASSWORD"]
```

File: acceptance/support-skeletons/synced_folder_smb/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  config.vm.synced_folder ".", "/foo", type: "smb", disabled: true
+  # Guest should persist mounts after reboot
+  config.vm.provision "shell", inline: "echo goodbye", reboot: true
```
Comment (jbonhag) [inline acceptance/support-skeletons/synced_folder_smb/Vagrantfile:7]: Maybe we want to set this provisioner to `run: "never"` and run it explicitly as part of the tests? That way we can check both of these scenarios:

1. Synced folder is mounted on the initial up
2. Synced folder persists after a reboot

**Review by jbonhag — DISMISSED** [review]
📁 🎉 Nice! Added a couple of ideas.

**Review by chrisroberts — APPROVED** [review]

**Comment by soapy1** [comment]
rebased and added bit to skip tests on windows, merging 🚒

**Pull request merged**